scipy/stats/_continuous_distns.py

```diff
@@ -5206,6 +5206,79 @@ def _entropy(self, beta):
 halfgennorm = halfgennorm_gen(a=0, name='halfgennorm')
 
 
+class histogram_gen(rv_continuous):
+    """
+    Generates a distribution given by a histogram.
+    This is useful to generate a template distribution from a binned datasample.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    There are no additional shape parameters except for the loc and scale.
+    The pdf and cdf are defined as stepwise functions from the provided histogram.
+    In particular the cdf is not interpolated between bin boundaries and not differentiable.
+
+    %(after_notes)s
+
+    %(example)s
+
+    data = scipy.stats.norm.rvs(size=100000, loc=0, scale=1.5)
+    hist = np.histogram(data, bins=100)
+    template = scipy.stats.histogram_gen(hist)
+
+    """
+    _support_mask = rv_continuous._support_mask
+
+    def __init__(self, histogram, *args, **kwargs):
+        """
+        Create a new distribution using the given histogram
+        @param histogram the return value of np.histogram
+        """
+        self.histogram = histogram
```

Comment:
`histogram` is a user-facing parameter, hence it needs to be documented in the class docstring, in the numpydoc format.  It's assumed to be a 2-tuple of 1D array-likes, where the first one is one longer then the second one, correct?
Also need to validate the input a bit. At least check the lengths, wrap `pdf`, `bins` into `np.asarray` in case they are lists etc.